src/core.ts

```diff
@@ -103,5 +101,5 @@ export function getSanitizedEnv<T>(
 
   const reporter = options?.reporter || defaultReporter
   reporter({ errors, env: cleanedEnv })
-  return cleanedEnv
+  return cleanedEnv as FromSpecsRecord<S>
```

Comment:
That one is really tricky because `FromSpecsRecord` use type inferences on `S`, upon which we have no type information in the body of this function (in this context, `S` is an alias for `unknown`). And for some reason, adding a type constraint on `S` breaks `cleanEnv` type inferences:

```ts
export function cleanEnv<S extends Record<string, ValidatorSpec<unknown>>>(

// ....
export function getSanitizedEnv<S extends Record<string, ValidatorSpec<unknown>>>(
```

![image](https://user-images.githubusercontent.com/3646758/202903127-315ee02a-870d-444d-886b-ecbaaf3779c2.png)

Reply:
No problem, was just wondering!

Reply:
Actually, I did found a trick: we don't need `FromSpecsRecord` to be so specific, `CleanedEnv` will do that. So I simplified `FromSpecsRecord`  definition, and renamed it to `SpecsOutput` in e38457ca78458cace183fedead33882ea958fe05.